h2o-core/src/main/java/water/fvec/InteractionWrappedVec.java

```diff
@@ -351,9 +351,19 @@ else if( _perChkMapMissing!=null && t._perChkMapMissing!=null ) {
     @Override public double atd_impl(int idx) {
       if( _isCat )
         if( isNA_impl(idx) ) return Double.NaN;
-      return _isCat ? Arrays.binarySearch(_vec.domain(), getKey(idx)) : ( _c1IsCat?1: (_c[0].atd(idx))) * ( _c2IsCat?1: (_c[1].atd(idx)) );
+       if (_isCat) {
+         int val = Arrays.binarySearch(_vec.domain(), getKey(idx));
+         return ((val < 0)?-1:val);
+       } else
```

Comment:
note that val = Arrays.binarySearch(_vec.domain(), getKey(idx)) can return bad values like -3.  Hence, I added the check to make sure we only return -1 for illegal values.